### ISSUE TITLE
feature request: return the 'final' parse patterns used for read in data (along with already returned col types)

### ISSUE THREAD

**Issue by statsccpr** [tidyverse/readr#728]
There is already the function `spec_csv()` which returns the column types.

But can there also be a function that returns the set of patterns used for delimiters, comments, nas, Nulls? "meta data" that would appease the wish list in the rant here
http://www.johnmyleswhite.com/notebook/2016/09/23/no-juice-for-you-csv-format-it-just-makes-you-more-awful/

say a list containing
quote = "\"", escape_backslash = FALSE,
  escape_double = TRUE, col_names = TRUE, col_types = NULL,
  locale = default_locale(), na = c("", "NA"), quoted_na = TRUE,
  comment = ""

The above 'meta data' can be taken from the default argument values say `args(spec_csv)`

Basically, i'm proposing `read_csv(file,verbose=TRUE)`
where `verbose=TRUE`, would output the meta data patterns along with the already output column types from`spec_csv()`

I guess you could instead push this into `spec_csv(,return_pattern=TRUE)`

I've kind of pushed the `data.table::fread` guys on this too, since there seems to be others with this feature wish

https://github.com/Rdatatable/data.table/issues/2437

**Comment by jimhester** [comment]
Fixed by #1172, which includes more of this metadata in the spec output, though there is still some that is not there.
